### PR TITLE
fix(web): update assets after delete

### LINKED ISSUE #8338: [Bug] Removing photo from search form doesn't show next photo
### The bug

When you delete an open photo from the search form, the photo does not change to the next one from the list, although it is actually deleted.

### The OS that Immich Server is running on

TrueNAS-SCALE-23.10.2

### Version of Immich Server

v1.99.0

### Version of Immich Mobile App

v1.99.0

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
https://github.com/truenas/charts/tree/master/community/immich
```

### Your .env content

```Shell
https://github.com/truenas/charts/tree/master/community/immich
```

### Reproduction steps

```bash
1. Go to the photo search, apply any filter
2. Open the photo in the filtered list of photos
3. Click "Delete", we will see the message "Moved to Trash" but the photo is still there
4. Return to the filtered photo list, the deleted photo is still in the list (but the photo is actually deleted)
5. When the window is refreshed, deleted photos disappear from the list. We can find the deleted photos in the Trash
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Xiayucheng1212**
fixes #8338 

## Current Behavior
After click the delete button on the viewer page, it will show the next photo by "navigateAsset('next')". Also, the assets shown in the gallery viewer will also filter out the deleted photo, and update itself.

### CODE REVIEW

**Review by jrasm91 — APPROVED** [review]

File: web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte [hunk 1]
```diff
@@ -65,6 +66,20 @@
     }
   };
 
+  const handleAction = (action: AssetAction, asset: AssetResponseDto) => {
+    switch (action) {
+      case AssetAction.DELETE:
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:71]: Should we handle the archive action as well like this?

Comment (Xiayucheng1212) [reply web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:71]: Yes, archieve is added. Do you think we should also navigate to next photo for action arvhieve?

Comment (Xiayucheng1212) [reply web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:72]: Yes, archieve is added. Do you think we should also navigate to next photo for action arvhieve?

Comment (jrasm91) [reply web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:72]: The idea is that if you are on the viewer and click "archive", "delete" or "trash" the asset no longer belongs to the current collection, so auto navigating forward seems to make a lot of sense. So yeah, I would have all of them to that same action.

Comment (jrasm91) [reply web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:73]: It turns out that it is better if the gallery component handles the navigation to next/previous, in order to not double up on events (the asset grid also does navigation on trash/delete - the parent component of the asset viewer in some situations)

**Review by danieldietzler — APPROVED** [review]

**Pull request merged**